src/main/kotlin/pt/up/fe/ni/website/backend/service/activity/AbstractActivityService.kt

```diff
@@ -83,4 +84,35 @@ abstract class AbstractActivityService<T : Activity>(
         activity.teamMembers.removeIf { it.id == idAccount }
         return repository.save(activity)
     }
+
+    fun addGalleryPhoto(activityId: Long, image: MultipartFile): Activity {
+        if (!repository.existsById(activityId)) {
+            throw NoSuchElementException(ErrorMessages.eventNotFound(activityId))
+        }
+
+        val activity = getActivityById(activityId)
+
+        val fileName = fileUploader.buildFileName(image, activity.title)
+        val imageName = fileUploader.uploadImage("gallery", fileName, image.bytes)
```

Comment:
Here it would be best to prefix this path with the `IMAGE_FOLDER` of each `Activity`, in order to separate them , no? Either that or just that same exact folder, what do you think?